### PR TITLE
Protect user profile routes

### PR DESCRIPTION
**Pull request by filipsobol**
### Short Description and Why It's Useful
Adds `authenticated.js` middleware and registers it in `MyAccount.vue` to protect user profile routes.

By default, middleware only emits an error message, asking to implement vendor-specific validation. For `commercetools` integration, checking if the user is logged in is based on cookies, because we can't use composables in middleware (might change when we update to Vue 3).

Additionally, I updated the default logger to use matching `console` log levels, like `debug`, `info`, `warn`, `error`.

### Which Environment This Relates To
* [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
* [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
* [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
* [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
* [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature
* [x] (Next only) I've followed this [instruction](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and I've created a `.js` file with information about my Pull Request

### CODE REVIEW

File: packages/core/docs/commercetools/changelog/5442.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
```
Comment (filrak) [inline packages/core/docs/commercetools/changelog/5442.js:1]: why its duplicated?

File: packages/commercetools/theme/middleware/authenticated.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default ({ app, redirect }) => {
+  if (!app.$cookies.get('vsf-commercetools-token')?.scope?.includes('customer_id')) {
```
[the remark this answers is not in the document]
Comment (filipsobol) [reply packages/commercetools/theme/middleware/authenticated.js]: Nice catch, updated

File: packages/core/docs/commercetools/changelog/5442.js (hunk 1 above)
Comment (filipsobol) [reply packages/core/docs/commercetools/changelog/5442.js:1]: There are 3 changelog files - 2 for `core` (they are unrelated, so I decided to split them into two) and this one for `commercetools`.

Comment (filrak) [reply packages/core/docs/commercetools/changelog/5442.js:1]: ah right, i overlooked this part, sorry ;)

File: packages/core/core/src/utils/logger/defaultLogger.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
 /* istanbul ignore file */
 
 const defaultLogger = {
-  debug: (message: any, ...args) => console.log('[VSF][debug]', message, ...args),
-  info: (message: any, ...args) => console.log('[VSF][info]', message, ...args),
-  warn: (message: any, ...args) => console.log('[VSF][warn]', message, ...args),
-  error: (message: any, ...args) => console.log('[VSF][error]', message, ...args)
+  debug: (message: any, ...args) => console.debug('[VSF][debug]', message, ...args),
```
Comment (andrzejewsky) [inline packages/core/core/src/utils/logger/defaultLogger.ts:4]: those calls don't work on Node, so please keep `log`

File: packages/commercetools/theme/middleware/authenticated.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default async ({ $vsf, redirect }) => {
+  if (await $vsf.$ct.api.isGuest()) {
```
Comment (andrzejewsky) [inline packages/commercetools/theme/middleware/authenticated.js:2]: remember that isGuest is true when you have an item in the cart or wishlist but you still not logged in

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: packages/commercetools/theme/middleware/authenticated.js (hunk 4 above)
Comment (filrak) [reply packages/commercetools/theme/middleware/authenticated.js:2]: is there a bug report for this?

File: packages/core/core/src/utils/logger/defaultLogger.ts (hunk 3 above)
Comment (filipsobol) [reply packages/core/core/src/utils/logger/defaultLogger.ts:4]: I tested it and it worked fine for me. Node docs suggest that it always supported these methods.

File: packages/commercetools/theme/middleware/authenticated.js (hunk 4 above)
Comment (filipsobol) [reply packages/commercetools/theme/middleware/authenticated.js]: fixed

**Review by filrak — APPROVED** [review]

File: packages/core/nuxt-theme-module/theme/pages/MyAccount.vue [hunk 5]
```diff
@@ -72,6 +72,9 @@ export default {
     OrderHistory,
     MyReviews
   },
+  middleware: [
+    'authenticated'
```
Comment (filrak) [inline packages/core/nuxt-theme-module/theme/pages/MyAccount.vue:76]:
```suggestion
    'is-authenticated'
```
?

File: packages/core/docs/changelog/5442_2.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  description: 'Updated the default logger to use matching console log levels, like debug, info, warn, error.',
+  link: 'https://github.com/vuestorefront/vue-storefront/pull/5442',
+  isBreaking: false,
```
Comment (filrak) [inline packages/core/docs/changelog/5442_2.js:4]:
```suggestion
  isBreaking: true,
```
it is a breaking change for the core

**Review by filrak — CHANGES_REQUESTED** [review]

File: packages/core/docs/changelog/5442_1.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  description: 'Added `is-authenticated` middleware to protect user profile routes from guest access. By default it emits an error message asking to implement vendor-specific validation.',
+  link: 'https://github.com/vuestorefront/vue-storefront/pull/5442',
+  isBreaking: true,
+  breakingChanges: [],
```
Comment (filrak) [inline packages/core/docs/changelog/5442_1.js:5]: missing before/after. If you include BC's please always put them in this table. It's easier to parse for integrators

File: packages/core/docs/changelog/5442_2.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  description: 'Updated the default logger to use matching console log levels, like debug, info, warn, error.',
+  link: 'https://github.com/vuestorefront/vue-storefront/pull/5442',
+  isBreaking: true,
```
Comment (filrak) [inline packages/core/docs/changelog/5442_2.js:4]:
```suggestion
  isBreaking: false,
```

**Pull request merged**